terraform/helm/aptos-node/files/haproxy.cfg

```diff
@@ -0,0 +1,123 @@
+global
+    log stdout len 10240 format raw local0
+    maxconn 500000
+    nbthread 16
+    user nobody
+
+defaults
+    log global
+    option tcplog
+    maxconn 500000
+    timeout queue 1s
+    timeout connect 10s
+    timeout server 60s
+    timeout client 60s
+    timeout client-fin 5s
+
+frontend validator
+    bind :6180
+    default_backend validator
+
+    # Deny requests from blocked IPs
+    tcp-request connection reject if { src -n -f /usr/local/etc/haproxy/blocked.ips }
+
+    # Limit to N TCP connections per minute per source IP
+    stick-table type ip size 500k expire 1m store gpc0_rate(1m)
+    tcp-request connection track-sc0 src
+    # TODO: Reject at content phase for now so we get logs, but this should be
+    # done at connection phase for higher efficiency
+    tcp-request content reject if { sc_gpc0_rate(0) ge {{ .Values.haproxy.limits.validator.connectionsPerIPPerMin }} }
+    tcp-request content sc-inc-gpc0(0) unless { nbsrv(validator) eq 0 }
+
+backend validator
+    default-server maxconn 1024 {{ if .Values.haproxy.config.send_proxy_protocol }}send-proxy-v2{{ end }}
+    server {{ include "aptos-validator.fullname" . }}-validator {{ include "aptos-validator.fullname" . }}-validator:6180
+
+frontend validator-metrics
+    mode http
+    option httplog
+    bind :9102
+    default_backend validator-metrics
+    http-request add-header Forwarded "for=%ci"
+
+    # Deny requests from blocked IPs
+    tcp-request connection reject if { src -n -f /usr/local/etc/haproxy/blocked.ips }
+
+backend validator-metrics
+    mode http
+    default-server maxconn 1024
+    server {{ include "aptos-validator.fullname" $ }}-validator {{ include "aptos-validator.fullname" $ }}-validator:9101
+
+# Exposes the validator's own REST API
+{{- if .Values.exposeValidatorRestApi }}
```

Comment:
this value isn't in `values.yaml`. might result in error applying the helm release

Reply:
maybe we can rename it to `.Values.validator.enableRestApi` ? think we should get rid of the top level helm values tbh

Reply:
EDIT: looks like `.Values.validator.exposeRestApi` already exists. we just need to combine them...